File: frontend/src/api/index/term.ts
import {get} from '/@/utils/http/axios';

enum URL {
    list = '/CSAA/index/term/list',
}

const listApi = async (params: any) =>
    get<any>({url: URL.list, params: params, data: {}, headers: {}});


export {listApi};

File: backend/CSAA/views/index/term.py
from rest_framework.decorators import api_view, authentication_classes

from CSAA import utils
from CSAA.auth.authentication import AdminTokenAuthtication
from CSAA.handler import APIResponse
from CSAA.models import Term
from CSAA.serializers import TermSerializer


# term列表
@api_view(['GET'])
def list_api(request):
    if request.method == 'GET':
        terms = Term.objects.all()
        serializer = TermSerializer(terms, many=True)
        return APIResponse(code=0, msg='查询成功', data=serializer.data)

